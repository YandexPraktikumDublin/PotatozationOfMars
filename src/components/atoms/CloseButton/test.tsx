import React from 'react'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'

import { CloseButton } from '.'

describe('<CloseButton />', () => {
  it('should renders correct <CloseButton />', () => {
    const onClick = jest.fn()

    const wrapper = shallow(<CloseButton onClick={onClick} />)

    wrapper.simulate('click')

    expect(onClick).toHaveBeenCalledTimes(1)

    expect(toJSON(wrapper)).toMatchSnapshot()
  })

  it('should has className from className prop', () => {
    const onClick = jest.fn()
    const className = 'test-class-name'

    const wrapper = shallow(
      <CloseButton onClick={onClick} className={className} />
    )

    expect(wrapper.prop('className')).toMatch(className)

    expect(toJSON(wrapper)).toMatchSnapshot()
  })
})
