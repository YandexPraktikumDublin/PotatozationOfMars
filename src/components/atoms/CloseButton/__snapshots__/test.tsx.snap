// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CloseButton /> should has className from className prop 1`] = `
<button
  className="test-class-name"
  onClick={[MockFunction]}
>
  <svg
    height="22"
    width="22"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      className="fill-primary dark:fill-white"
      d="M21 21L1 1M21 1L1 21"
      strokeWidth="2"
    />
  </svg>
</button>
`;

exports[`<CloseButton /> should renders correct <CloseButton /> 1`] = `
<button
  onClick={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
>
  <svg
    height="22"
    width="22"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      className="fill-primary dark:fill-white"
      d="M21 21L1 1M21 1L1 21"
      strokeWidth="2"
    />
  </svg>
</button>
`;
